### PR TITLE
Completed User module

### LINKED ISSUE #20: make User controller
Make controllers for the user entity

### PR DESCRIPTION
**Pull request by ritvij14**
## Pull request checklist

Please check if your PR :rocket: fulfills the following requirements:
- [x] Build (`npm run start`) was run locally and new changes were working correctly
- [x] Lint (`npm run linter`) has passed locally and any fixes were made for failures

## Pull request type

 

Please check the type of change your PR introduces:
- [x] Feature

Fixes #20 

## What is the new behavior?

Screenshots of Postman results and database:-
### Create user
WhatsApp Image 2021-03-26 at 6 21 10 PM
WhatsApp Image 2021-03-26 at 6 30 39 PM

### Get user (by id)
image

### Update user
image
image
image

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/modules/User/user.service.ts [hunk 1]
```diff
@@ -14,12 +15,18 @@ export class UserService {
 	}
 
 	@Transactional()
-    async createUser(user: Users): Promise<Users> {
+	async createUser(user: CreateUserDto): Promise<Users> {
```
Comment (Sarvesh67) [inline src/modules/User/user.service.ts]: Don't use DTO here, use it only in the controller and convert the input into Users object in the controller itself.

**Review by Sarvesh67 — CHANGES_REQUESTED** [review]

File: src/modules/User/user.service.ts (hunk 1 above)
Comment (ritvij14) [reply src/modules/User/user.service.ts]: Done

**Review by Sarvesh67 — CHANGES_REQUESTED** [review]
- Do not use error handling in the controller itself. Apply exception filters at either the controller level or the handler level depending on its relevance.
 - Add a delete user api too.

File: src/modules/User/user.service.ts [hunk 2]
```diff
@@ -19,7 +37,37 @@ export class UserService {
 	}
 
 	@Transactional()
-	async findByEmail(email: string): Promise<Users | undefined> {
-		return await this.userRepo.findOne({ email: email });
+	async updateUser(userID: number, user: Users): Promise<Users | undefined> {
```
Comment (Sarvesh67) [inline src/modules/User/user.service.ts:40]: Remove findById method calls. Just use the response of update method.

**Review by Sarvesh67 — CHANGES_REQUESTED** [review]

**Pull request merged**